src/Microsoft.VisualStudio.SDK.Analyzers/VSSDK013ThreadAffinitizedMEFConstruction.cs

```diff
@@ -0,0 +1,226 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Immutable;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.VisualStudio.SDK.Analyzers
+{
+    /// <summary>
+    /// Identifies cases where a class decorated with [Export] accesses any member bound to UI thread from
+    /// - Constructor with no params
+    /// - Constructor decorated with [ImportingConstructor]
+    /// - Field or property initializer
+    /// - implementation of IPartImportsSatisfiedNotification.OnImportsSatisfied.
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class VSSDK013ThreadAffinitizedMEFConstruction : DiagnosticAnalyzer
+    {
+        /// <summary>
+        /// The value to use for <see cref="DiagnosticDescriptor.Id"/> in generated diagnostics.
+        /// </summary>
+        public const string Id = "VSSDK013";
+
+        /// <summary>
+        /// A reusable descriptor for diagnostics produced by this analyzer.
+        /// </summary>
+        internal static readonly DiagnosticDescriptor Descriptor = new DiagnosticDescriptor(
+            id: Id,
+            title: "Free threaded MEF Part construction",
+            messageFormat: "MEF Parts construction should not have UI thread affinity",
+            helpLinkUri: Utils.GetHelpLink(Id),
+            category: "Reliability",
+            defaultSeverity: DiagnosticSeverity.Warning,
+            isEnabledByDefault: true);
+
+        private const RegexOptions FileNamePatternRegexOptions = RegexOptions.IgnoreCase | RegexOptions.Singleline;
```

Comment:
This is part of `CommonInterest` in the `vs-threading` repo. What's the best way to use these helpers in this repo? So far, I've copy-pasted relevant code to this type, but I would like to align with your vision for this repo.